tools/objc/libtool_check_unique.cc

```diff
@@ -32,43 +37,71 @@ string getBasename(const string &path) {
   return path.substr(idx + 1);
 }
 
-// Returns 0 if there are no duplicate basenames in the object files (both via
-// -filelist as well as shell args), 1 otherwise
-int main(int argc, const char *argv[]) {
+std::vector<string> readFile(const string path) {
+  std::vector<string> lines;
+  ifstream file(path);
+  string line;
+  while (std::getline(file, line)) {
+    if (!line.empty()) {
+      lines.push_back(line);
+    }
+  }
+
+  return lines;
+}
+
+unordered_set<string> parseArgs(std::vector<string> args) {
   unordered_set<string> basenames;
-  const regex libRegex = regex(".*\\.a$");
-  const regex noArgFlags =
-      regex("-static|-s|-a|-c|-L|-T|-D|-no_warning_for_no_symbols");
-  const regex singleArgFlags = regex("-arch_only|-syslibroot|-o");
-  // Set i to 1 to skip executable path
-  for (int i = 1; argv[i] != nullptr; i++) {
-    const string arg = argv[i];
+  for (auto it = args.begin(); it != args.end(); ++it) {
+    const string arg = *it;
     if (arg == "-filelist") {
-      ifstream list(argv[i + 1]);
+      ifstream list(arg);
       for (string line; getline(list, line);) {
         const string basename = getBasename(line);
         const auto pair = basenames.insert(basename);
         if (!pair.second) {
-          return EXIT_FAILURE;
+          exit(EXIT_FAILURE);
         }
       }
       list.close();
-      i++;
+      ++it;
+    } else if (arg[0] == '@') {
+      string paramsFilePath(arg.substr(1));
+      auto newBasenames = parseArgs(readFile(paramsFilePath));
+      for (auto newBasename : newBasenames) {
+        const auto pair = basenames.insert(newBasename);
+        if (!pair.second) {
+          exit(EXIT_FAILURE);
+        }
+      }
     } else if (regex_match(arg, noArgFlags)) {
     } else if (regex_match(arg, singleArgFlags)) {
-      i++;
+      ++it;
     } else if (arg[0] == '-') {
-      return EXIT_FAILURE;
+      exit(EXIT_FAILURE);
       // Unrecognized flag, let the wrapper deal with it
```

Comment:
unrelated, but thinking more about this case, i wonder what the odds are of flags being added (or have they been already?) in which case this would silently become a performance regression. i wonder if it makes sense to output a line somewhere saying libtool has to fall back to the slow path

Reply:
ah but looking at this again, it looks like we're confined to a specific set of args anyway or it errors out

Reply:
yea so whenever libtool.sh is updated to handle new args, we just have to remember to do it here too

Reply:
May be worth adding a comment to that effect?